x/gamm/pool-models/balancer/tx.pb.go

```diff
@@ -188,72 +188,72 @@ var _ grpc.ClientConn
+func (c *balancerMsgClient) CreateBalancerPool(ctx context.Context, in *MsgCreateBalancerPool, opts ...grpc.CallOption) (*MsgCreateBalancerPoolResponse, error) {
 	out := new(MsgCreateBalancerPoolResponse)
-	err := c.cc.Invoke(ctx, "/osmosis.gamm.v1beta1.Msg/CreateBalancerPool", in, out, opts...)
+	err := c.cc.Invoke(ctx, "/osmosis.gamm.v1beta1.BalancerMsg/CreateBalancerPool", in, out, opts...)
 	if err != nil {
 		return nil, err
 	}
 	return out, nil
 }
 
-// MsgServer is the server API for Msg service.
-type MsgServer interface {
+// BalancerMsgServer is the server API for BalancerMsg service.
+type BalancerMsgServer interface {
 	CreateBalancerPool(context.Context, *MsgCreateBalancerPool) (*MsgCreateBalancerPoolResponse, error)
 }
 
-// UnimplementedMsgServer can be embedded to have forward compatible implementations.
-type UnimplementedMsgServer struct {
+// UnimplementedBalancerMsgServer can be embedded to have forward compatible implementations.
+type UnimplementedBalancerMsgServer struct {
 }
 
-func (*UnimplementedMsgServer) CreateBalancerPool(ctx context.Context, req *MsgCreateBalancerPool) (*MsgCreateBalancerPoolResponse, error) {
+func (*UnimplementedBalancerMsgServer) CreateBalancerPool(ctx context.Context, req *MsgCreateBalancerPool) (*MsgCreateBalancerPoolResponse, error) {
 	return nil, status.Errorf(codes.Unimplemented, "method CreateBalancerPool not implemented")
 }
 
-func RegisterMsgServer(s grpc1.Server, srv MsgServer) {
-	s.RegisterService(&_Msg_serviceDesc, srv)
+func RegisterBalancerMsgServer(s grpc1.Server, srv BalancerMsgServer) {
+	s.RegisterService(&_BalancerMsg_serviceDesc, srv)
 }
 
-func _Msg_CreateBalancerPool_Handler(srv interface{}, ctx context.Context, dec func(interface{}) error, interceptor grpc.UnaryServerInterceptor) (interface{}, error) {
+func _BalancerMsg_CreateBalancerPool_Handler(srv interface{}, ctx context.Context, dec func(interface{}) error, interceptor grpc.UnaryServerInterceptor) (interface{}, error) {
 	in := new(MsgCreateBalancerPool)
 	if err := dec(in); err != nil {
 		return nil, err
 	}
 	if interceptor == nil {
-		return srv.(MsgServer).CreateBalancerPool(ctx, in)
+		return srv.(BalancerMsgServer).CreateBalancerPool(ctx, in)
 	}
 	info := &grpc.UnaryServerInfo{
 		Server:     srv,
-		FullMethod: "/osmosis.gamm.v1beta1.Msg/CreateBalancerPool",
+		FullMethod: "/osmosis.gamm.v1beta1.BalancerMsg/CreateBalancerPool",
 	}
 	handler := func(ctx context.Context, req interface{}) (interface{}, error) {
-		return srv.(MsgServer).CreateBalancerPool(ctx, req.(*MsgCreateBalancerPool))
+		return srv.(BalancerMsgServer).CreateBalancerPool(ctx, req.(*MsgCreateBalancerPool))
 	}
 	return interceptor(ctx, in, info, handler)
 }
 
-var _Msg_serviceDesc = grpc.ServiceDesc{
-	ServiceName: "osmosis.gamm.v1beta1.Msg",
-	HandlerType: (*MsgServer)(nil),
+var _BalancerMsg_serviceDesc = grpc.ServiceDesc{
+	ServiceName: "osmosis.gamm.v1beta1.BalancerMsg",
```

Comment:
Is this service name at all important? 

an alternative approach to fixing this problem entirely, is to migrate the balancer pools and not deal with this duplicate import situation

Reply:
why is still it creating BalancerMsg under v1beta1?  I was assuming it should be created as something like `osmosis.gamm.pool-models.BalacerMsg`

Wdym by migrating balancer pools?

Reply:
oh thats a good point, we could make txs under the correct package, but just leave the pool proto definition under the legacy wrong one

Reply:
I think your proposal is a better approach